scripts/update_and_rebuild_libmesh.sh

```diff
@@ -56,12 +56,10 @@ $SCRIPT_DIR/diagnostics.sh > "$SCRIPT_DIR/$DIAGNOSTIC_LOG"
 
 if [[ -n "$LIBMESH_DIR" ]]; then
   echo "INFO: LIBMESH_DIR set - overriding default installed path"
-  echo "INFO: No cleaning will be done in specified path"
-  mkdir -p $LIBMESH_DIR
 else
   export LIBMESH_DIR=$SCRIPT_DIR/../libmesh/installed
-  rm -rf $SCRIPT_DIR/../libmesh/installed
 fi
+rm -rf $LIBMESH_DIR
```

Comment:
Wait! Are you sure? If I have LIBMESH_DIR set it just deletes it?

Reply:
Sure, why not? Aren't we rebuilding? We were previously running this command in the default location before rebuilding. Why wouldn't we do it for any location?

Reply:
The issue was (maybe not is anymore) that developers use to have a separate libmesh directory that they had in progress development inside. This wasn't a submodule, but a real checkout of the library. Running this script with LIBMESH_DIR defined was generally an accident and you didn't want to blow away your main repository so we handeld that case.

Imagine today with all the copies of MOOSE that you might have on your system if we blew away MOOSE_DIR when building an application just because that variable was set? The same argument applies. I might want several different MOOSE directories but I don't necessarily want you to clobber my main repos because I forgot to unset the variable.

Reply:
Developers would set LIBMESH_DIR equal to the root of their libmesh checkout? So they'd just be adding bin, include, lib directories alongside the source? I don't see how that would work since libmesh root already has an include directory for example.

Reply:
I guess Derek does do in-tree builds...